lib/auth/auth.go

```diff
@@ -2318,10 +2340,20 @@ func (a *Server) NewWebSession(req types.NewWebSessionRequest) (types.WebSession
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	checker, err := services.FetchRoles(req.Roles, a.Access, req.Traits)
+	roleSet, err := services.FetchRoles(req.Roles, a.Access, req.Traits)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	clusterName, err := a.GetDomainName()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	checker := services.NewAccessChecker(&services.AccessInfo{
+		Roles:              req.Roles,
+		Traits:             req.Traits,
+		RoleSet:            roleSet,
+		AllowedResourceIDs: req.RequestedResourceIDs,
```

Comment:
Should you be using `req.RequestedResourceIDs` or `allowedResourceIDs` here?

Having the separate var that you populate and then re-set to nil confuses me, and it feels like it will be too easy to introduce a regression by mistakenly using the wrong set of IDs here.

Reply:
My bad, this is a case of me not interpreting the GitHub blue line as a break in the file.